driver/SDL/sdl_common.c

```diff
@@ -73,20 +74,13 @@ void sdl_mousewheel_read(lv_indev_drv_t * indev_drv, lv_indev_data_t * data)
 void sdl_keyboard_read(lv_indev_drv_t * indev_drv, lv_indev_data_t * data)
 {
     (void) indev_drv;      /*Unused*/
-
-    static bool dummy_read = false;
     const size_t len = strlen(buf);
 
-    /*Send a release manually*/
-    if (dummy_read) {
-        dummy_read = false;
-        data->state = LV_INDEV_STATE_RELEASED;
-        data->continue_reading = len > 0;
-    }
-        /*Send the pressed character*/
-    else if (len > 0) {
-        dummy_read = true;
+    /*Send data*/
+    if (len > 0) {
         data->state = LV_INDEV_STATE_PRESSED;
+        if(key_released)
```

Comment:
How do you know *which* key was released?
Do you perform `memmove` on both "press" and "release" events? 

I think that if we want to handle press/release states individually, we need a fifo of events where each event consists of both the character-code and the state ("pressed"/"released").   
Today we only have a character buffer that is used as a fifo (with the memmove), but does not record the state of each character.

Reply:
I have reimplemented it by using a FIFO struct array. Now the order of events is considered correctly.